src/Airship/Internal/Helpers.hs

```diff
@@ -22,12 +23,25 @@ import qualified Network.HTTP.Types        as HTTP
 import qualified Network.Wai               as Wai
 import           Network.Wai.Parse
 import           System.Random
+import           Text.Read
 
+import           Airship.Headers
 import           Airship.Internal.Decision
 import           Airship.Internal.Route
 import           Airship.Resource
 import           Airship.Types
-import           Airship.Headers
+
+-- | Reads the parameter of the provided name and attempt to convert
+-- it to the desired 'Read'-implementing type. If the parameter is not
+-- found or cannot be read successfully, the provided handler is invoked
+-- (this may often be `halt status404`).
+readParam :: Read a => Text -> Handler s m a -> Handler s m a
```

Comment:
Welcome @charleso!

`readParam` would be used for reading query strings from a request. So, a common use-case might be something like:

```
GET /foo/bar?enabled=True
```

```haskell
enabled :: Maybe Bool <- readParamMaybe "enabled"
```

So just the _name_ of the query-string is `Text`. As you've correctly pointed out though, using the `Read` type-class for this can be brittle. Case in point, `True` parses as the Haskell `Bool` `True`, but `true` does not. I would consider this a convenience helper, and not something I would necessary use for production applications that need to be more solid.